apps/crud/src/app/app.component.ts

```diff
@@ -1,51 +1,94 @@
-import { HttpClient } from '@angular/common/http';
-import { Component, OnInit } from '@angular/core';
-import { randText } from '@ngneat/falso';
+import { Component, OnInit, ChangeDetectionStrategy } from '@angular/core';
+import { TodoConfig } from './core/Interface/todo';
+import { GetToDoService } from './services/getTodo.service';
+import { Observable, of } from 'rxjs';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
+import { ItemComponent } from './shared/item.component';
 
 @Component({
   standalone: true,
-  imports: [CommonModule],
+  imports: [CommonModule, MatProgressSpinnerModule, ItemComponent],
   selector: 'app-root',
+  changeDetection: ChangeDetectionStrategy.OnPush,
   template: `
-    <div *ngFor="let todo of todos">
-      {{ todo.title }}
-      <button (click)="update(todo)">Update</button>
+    <div class="container">
+      <div class="spinner" *ngIf="globalLoader | async; else showData">
+        <mat-spinner></mat-spinner>
+      </div>
+      <ng-template #showData>
+        <div class="show-view" *ngFor="let todo of viewData$ | async">
+          <app-item
+            [config]="todo"
+            [spinnerState]="isShowSpinner"
+            (deleteTodo)="delete($event)"
+            (updateTodo)="update($event)">
+          </app-item>
+        </div>
+      </ng-template>
     </div>
   `,
-  styles: [],
+  styles: [
+    `
+      html,
+      body {
+        height: 100vh;
+      }
+      .container {
+        position: relative;
+        display: flex;
+        flex-direction: column;
+        height: 100%;
+      }
+      .show-view {
+        border: 1px solid gray;
+      }
+      .title-view {
+        text-align: center;
+        font-size: 20px;
+      }
+      .button-view {
+        display: flex;
+        flex-direction: row;
+        justify-content: center;
+      }
+      button {
+        width: 80px;
+        height: 30px;
+        background-color: green;
+        color: white;
+        font-size: 20px;
+        margin: 5px 10px;
+      }
+      .spinner {
+        position: absolute;
+        display: flex;
+        flex-direction: column;
+        align-items: center;
+        height: 50%;
+        width: 100%;
+        top: 50%;
+      }
+    `,
+  ],
 })
 export class AppComponent implements OnInit {
-  todos!: any[];
-
-  constructor(private http: HttpClient) {}
+  viewData$ = this.getTodoService.data$;
+  isShowSpinner: Observable<boolean> = of(false);
+  globalLoader: Observable<boolean> = of(false);
+  isFirstTimeLoading = false;
+  constructor(private getTodoService: GetToDoService) {}
 
   ngOnInit(): void {
-    this.http
-      .get<any[]>('https://jsonplaceholder.typicode.com/todos')
-      .subscribe((todos) => {
-        this.todos = todos;
-      });
+    this.getTodoService.getTodoData();
+    this.globalLoader = this.getTodoService.globalLoader$;
+    this.isShowSpinner = this.getTodoService.localLoader$;
```

Comment:
re-initaliazing an observable is not a very good practise. You should initialized your observable directly l.77 and l.78

Reply:
done